gettsim/transfers/grundrente.py

```diff
@@ -0,0 +1,391 @@
+
+    return out.round(2)
+
+
+def grundr_zuschlag_vor_eink_anr_m(
+    bonus_entgeltpunkte_grundr: FloatSeries,
+    gr_bewertungszeiten: IntSeries,
+    rentenwert: FloatSeries,
+    zugangsfaktor: FloatSeries,
+    ges_renten_vers_params: dict,
+) -> FloatSeries:
+    """Calculate additional monthly pensions payments resulting from
+    Grundrente, without taking into account income crediting rules.
+
+    According to the Grundrentengesetz, the Zugangsfaktor is limited to 1
+    and considered Grundrentezeiten are limited to 35 years (420 months).
+    Parameters
+    ----------
+    bonus_entgeltpunkte_grundr
+        See :func:`bonus_entgeltpunkte_grundr`.
+    gr_bewertungszeiten
+        See basic input variable
+        :ref:`gr_bewertungszeiten <gr_bewertungszeiten>`.
+    rentenwert
+        See :func:`rentenwert`.
+    zugangsfaktor
+        See :func:`zugangsfaktor`.
+    ges_renten_vers_params
+        See params documentation :ref:`ges_renten_vers_params <ges_renten_vers_params>`.
+
+    Returns
+    -------
+
+    """
+    out = (
+        bonus_entgeltpunkte_grundr
+        * gr_bewertungszeiten.clip(
+            upper=ges_renten_vers_params["grundrentenzeiten"]["max"]
+        )
+        * rentenwert
+        * zugangsfaktor.clip(upper=1)
+    )
+    return out.round(2)
+
+
+def durchschnittl_entgeltpunkte_grundr(
+    entgeltp_grundr: FloatSeries, gr_bewertungszeiten: IntSeries
+) -> FloatSeries:
+    """Compute average number of Entgeltpunkte earned per month of Grundrentenzeiten.
+
+    Parameters
+    ----------
+    entgeltp_grundr
+        See basic input variable
+        :ref:`entgeltp_grundr <entgeltp_grundr>`.
+    gr_bewertungszeiten
+        See basic input variable
+        :ref:`gr_bewertungszeiten <gr_bewertungszeiten>`.
+
+    Returns
+    -------
+
+    """
+    return entgeltp_grundr / gr_bewertungszeiten
+
+
+def höchstwert_grundr_m(
+    grundrentenzeiten: IntSeries, ges_renten_vers_params: dict
+) -> FloatSeries:
+    """Calculates the maximum allowed number of average Entgeltpunkte (per month)
+    after adding bonus of Entgeltpunkte for a given number of Grundrentenzeiten.
+
+    Parameters
+    ----------
+    grundrentenzeiten
+        See basic input variable :ref:`grundrentenzeiten <grundrentenzeiten>`.
+    ges_renten_vers_params
+        See params documentation :ref:`ges_renten_vers_params <ges_renten_vers_params>`.
+
+    Returns
+    -------
+
+    """
+
+    # Calculate number of months above minimum threshold
+    months_above_thresh = (
+        grundrentenzeiten.clip(upper=ges_renten_vers_params["grundrentenzeiten"]["max"])
+        - ges_renten_vers_params["grundrentenzeiten"]["min"]
+    )
+
+    # Calculate höchstwert
+    out = (
+        ges_renten_vers_params["höchstwert_grundrente"]["base"]
+        + ges_renten_vers_params["höchstwert_grundrente"]["increment"]
+        * months_above_thresh
+    )
+
+    # Round to 4 digits
```

Comment:
You probably have an internal standard now how to implement rounding?We had an earlier discussion in #88 .

Reply:
As far as I understand, we are rounding whereever this is what is written in the law. This is the case here.

Thank you for the reminder to the issue -- I think it would be a good idea to make rounding optional at some point.

Reply:
Just the other way round AFAIR. The law often requires rounding to the nearest Euro and we decided against it because it might lead to discontinuities in otherwise smooth functions. IZAΨMOD behaved differently and did the rounding.

See #88.

Reply:
Sorry, for missrepresenting the conclusion in #88 then! Let's continue the discussion over there.